client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -0,0 +1,369 @@
+ * @returns rpaParameters as Array
+ */
+const getRpaParameters = (currentLine, splitPlaceholder, instructionBlocks) => {
+    const parametersWithoutRpaTask = currentLine.replace(instructionBlocks[instructionBlocks.length - 1].rpaTask + splitPlaceholder, '')
+    return parametersWithoutRpaTask.split([splitPlaceholder]);
+}
+
+/**
+ * @description "preprocesses" the code in a usable data format
+ * @param {Array} robotCodeTaskSection robot code w/o empty lines as an array of Strings
+ * @param {Array} declaredApplications all declared Aplications from ***settings*** section as Strings 
+ * @returns Array of Objects with the following schema:
+ *      instructionBlocks = [rpaApplication:String, rpaTask:String, name:String, paramArray:Array]
+ */
+const getInstructionBlocksFromTaskSection = (robotCodeTaskSection, declaredApplications) => {
+    let currentApplication;
+    let errorWasThrown;
+    const instructionBlocks = [];
+    const regexForElementNameLine = new RegExp(`#`)
+    const splitPlaceholder = '§&§';
+
+    robotCodeTaskSection.slice(1).forEach((line) => {
+        if (errorWasThrown) return;
+        let currentLine = line.trim();
+        const currentLineDefinesNewApplication = declaredApplications.includes(currentLine);
+        const currentLineContainsElementName = regexForElementNameLine.test(currentLine);
+        const currentLineIncludesSplitPlaceholder = currentLine.includes(splitPlaceholder);
+        const currentLineHasNoSpecifiedApplication = typeof currentApplication === 'undefined';
+
+        if (currentLineDefinesNewApplication) {
+            currentApplication = currentLine;
+            return;
+
+        } if (currentLineHasNoSpecifiedApplication) {
+            customNotification('Error', `There is no RPA-Application specified for line "${currentLine}"`)
+            errorWasThrown = true;
+            return;
+        }
+
+        if (currentLineIncludesSplitPlaceholder) {
+            customNotification('Error', `It is not allowed to use & or § as param values \nError location: "${line}"`)
+            errorWasThrown = true;
+            return;
+        }
+
+        if (currentLineContainsElementName) {
+            instructionBlocks.push({ rpaApplication: currentApplication, name: currentLine.substring(1) })
+            return;
+        }
+
+        if (!errorWasThrown) {
+            currentLine = currentLine.replaceAll(FOURSPACE, splitPlaceholder);
+            instructionBlocks[instructionBlocks.length - 1].rpaTask =
+                getRpaTask(currentLine, splitPlaceholder);
+            instructionBlocks[instructionBlocks.length - 1].paramArray =
+                getRpaParameters(currentLine, splitPlaceholder, instructionBlocks);
+
+            // TODO: implement parsing of outputVariables from RPAf code
+        }
+    })
+    return (errorWasThrown ? undefined : instructionBlocks);
+}
+
+/**
+ * @returns dummy startMarker as JSON => currently MARKERS aren't defined
```

Comment:
Should still feature a description if you ask me

Reply:
Unfortunately, I did not think of any content that is not already in the @return part. Therefore I have skipped that on purpose. But I add the following:
```suggestion
 * @description builds a dummy startMarker and returns them
 * @returns dummy startMarker as JSON => currently MARKERS aren't defined
```

Reply:
`@description Builds` capitalisation please

Reply:
Check :)